xdsl/dialects/arith.py

```diff
@@ -669,6 +732,23 @@ def get(operand1: Union[Operation, SSAValue],
                           result_types=[operand1.typ])
 
 
+@irdl_op_definition
+class Negf(Operation):
+    name: str = "arith.negf"
+    fastmath: OptOpAttr[FastMathFlagsAttr]
+    operand: Annotated[Operand, floatingPointLike]
+    result: Annotated[OpResult, floatingPointLike]
+
+    @staticmethod
+    def get(operand: Union[Operation, SSAValue],
+      fastmath: FastMathFlagsAttr | None = None) -> Negf:
+        attributes = {"fastmath": fastmath}
+
+        operand = SSAValue.get(operand)
```

Comment:
The _operand = SSAValue.get(operand)_ line? I think I do, as I am grabbing operand.typ for the result type and if I pass in an operand rather than SSA value it doesn't like it?